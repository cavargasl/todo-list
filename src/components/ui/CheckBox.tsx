import { cn } from "@/utils"

type CheckBoxProps = React.InputHTMLAttributes<HTMLInputElement>
export default function CheckBox({ className, type, ...props }: CheckBoxProps) {
  if (type) {
    console.warn(
      "The prop `type` is not supported on `CheckBox component`. always set `type` to `checkbox`."
    )
  }
  return (
    <input
      type="checkbox"
      className={cn(
        "checkbox h-5 w-5 cursor-pointer appearance-none rounded-full border bg-black px-2 outline-none checked:border-transparent checked:bg-green-500 focus:border-inherit focus:outline-none",
        className
      )}
      {...props}
    />
  )
}
