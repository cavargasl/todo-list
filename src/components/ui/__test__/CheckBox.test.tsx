import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import CheckBox from "../CheckBox"

describe("CheckBox", () => {
  test("renders correctly", () => {
    const { container } = render(<CheckBox />)
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveAttribute("type", "checkbox")
  })

  test("renders with custom className", () => {
    const { container } = render(
      <CheckBox className="size-10 px-4 checked:bg-pink-500" />
    )
    expect(container).toMatchSnapshot()
  })

  test("renders with additional props", () => {
    const { container } = render(
      <CheckBox aria-label="checkbox" defaultChecked disabled />
    )
    expect(container).toMatchSnapshot()
  })

  it("should change the checked state when clicked", async () => {
    render(<CheckBox />)
    const checkbox = screen.getByRole("checkbox")
    expect(checkbox).not.toBeChecked()
    await userEvent.click(checkbox)
    expect(checkbox).toBeChecked()
  })
})
