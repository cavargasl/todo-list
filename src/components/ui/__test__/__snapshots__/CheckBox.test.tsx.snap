// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CheckBox > renders correctly 1`] = `
<div>
  <input
    class="checkbox h-5 w-5 cursor-pointer appearance-none rounded-full border bg-black px-2 outline-none checked:border-transparent checked:bg-green-500 focus:border-inherit focus:outline-none"
    type="checkbox"
  />
</div>
`;

exports[`CheckBox > renders with additional props 1`] = `
<div>
  <input
    aria-label="checkbox"
    checked=""
    class="checkbox h-5 w-5 cursor-pointer appearance-none rounded-full border bg-black px-2 outline-none checked:border-transparent checked:bg-green-500 focus:border-inherit focus:outline-none"
    disabled=""
    type="checkbox"
  />
</div>
`;

exports[`CheckBox > renders with custom className 1`] = `
<div>
  <input
    class="checkbox cursor-pointer appearance-none rounded-full border bg-black outline-none checked:border-transparent focus:border-inherit focus:outline-none size-10 px-4 checked:bg-pink-500"
    type="checkbox"
  />
</div>
`;
